low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +39,133 @@ class ServerMap(ResourceMap):
 
     def __init__(self):
         """Initialize server"""
-        pass
+        super().__init__()
+        self.sysfs = ToolFactory().get_instance('sysfs')
+        self.sysfs.initialize()
+        self.sysfs_base_path = self.sysfs.get_sysfs_base_path()
+        self.cpu_path = self.sysfs_base_path + "devices/system/cpu/"
+
+        self.server_frus = {
+            'cpu': self.get_cpu_info
+        }
 
     def get_health_info(self, rpath):
         """
         Get health information of fru in given resource id
 
         rpath: Resource id (Example: nodes[0]>compute[0]>hw>disks)
         """
-        pass
+        info = {}
+        fru_found = False
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+        if leaf_node == "compute":
+            for fru in self.server_frus:
+                info.update({fru: self.server_frus[fru]()})
+            info["last_updated"] = int(time.time())
+            fru_found = True
+        else:
+            fru = None
+            for node in nodes:
+                fru, _ = self.get_node_details(node)
+                if self.server_frus.get(fru):
+                    fru_found = True
+                    info = self.server_frus[fru]()
+                    break
+        if not fru_found:
+            raise ResourceMapError(
+                errno.EINVAL,
+                "Health provider doesn't have support for'{rpath}'.")
+        return info
+
+    @staticmethod
+    def get_node_details(node):
+        """
+        Parse node information and returns left string and instance.
+
+        Example
+            "storage"    -> ("storage", "*")
+            "storage[0]" -> ("storage", "0")
+        """
+        res = re.search(r"(\w+)\[([\d]+)\]|(\w+)", node)
+        inst = res.groups()[1] if res.groups()[1] else "*"
+        node = res.groups()[0] if res.groups()[1] else res.groups()[2]
+        return node, inst
+
+    @staticmethod
+    def get_data_template(uid, is_fru: bool):
+        """Returns health template."""
+        return {
+            "uid": uid,
+            "fru": str(is_fru).lower(),
+            "last_updated": int(time.time()),
+            "health": {
+                "status": "",
+                "description": "",
+                "recommendation": "",
+                "specifics": []
+            }
+        }
```

Comment:
This is generic. We can put this in ResourceMap class.

Reply:
@mariyappanp I have added this function in `ResourceMap` class, please check.